### PR TITLE
[ENH] Make transformer files private

### LINKED ISSUE #2068: [ENH] Make all collection and series transformer files private, access through init
### Describe the feature or idea you want to propose

currently like this
image
should be like series transformers 
image

### Describe your proposed solution

make files private, import through init

### PR DESCRIPTION
**Pull request by TonyBagnall**
#### Reference Issues/PRs
fixes #2068 

- Makes remaining collection transformers private files
- Removes the term Panel from all docstrings, replaced by Collection
- Removes redundant code and docstrings from BaseTransformer
- adds some tests for coverage for incorrect input of parameters

### CODE REVIEW

**Review by MatthewMiddlehurst — COMMENTED** [review]
Fine with the refactor. Not really a fan of `test_wrong_paras.py`, too general. Think it would be better as class/file specific test files.

**Comment by TonyBagnall** [comment]
> Fine with the refactor. Not really a fan of `test_wrong_paras.py`, too general. Think it would be better as class/file specific test files.

will result in a lot of very small test files, but ok will split it

File: aeon/transformations/collection/base.py [hunk 1]
```diff
@@ -376,21 +373,3 @@ def _update(self, X, y=None):
...
-            )
-        if isinstance(y, np.ndarray) and y.ndim > 1:
-            raise TypeError(f"y must be 1-dimensional, found {y.ndim} dimensions")
-        # Check matching number of labels
-        n_labels = y.shape[0]
-        if n_cases != n_labels:
-            raise ValueError(
-                f"Mismatch in number of cases. Number in X = {n_cases} nos in y = "
-                f"{n_labels}"
-            )
```
Comment (MatthewMiddlehurst) [inline aeon/transformations/collection/base.py:396]: Why?

File: aeon/transformations/collection/_segment.py [hunk 2]
```diff
@@ -60,10 +60,6 @@ class IntervalSegmenter(BaseCollectionTransformer):
         and the second column giving end points of intervals
     """
 
-    _tags = {
-        "capability:unequal_length:removes": True,
-    }
```
Comment (MatthewMiddlehurst) [inline aeon/transformations/collection/_segment.py:65]: does it not do this?

File: aeon/transformations/base.py [hunk 3]
```diff
@@ -514,8 +306,9 @@ def _inverse_transform(self, X, y=None):
         X: data structure of type X_inner_type
             if X_inner_type is list, _inverse_transform must support all types in it
             Data to be transformed
-        y : Series or Panel, default=None
-            Additional data, e.g., labels for transformation
+        y : Series, default=None
+            Additional data, e.g., labels for transformation.
+        `
```
Comment (MatthewMiddlehurst) [inline aeon/transformations/base.py:311]: typo

File: aeon/transformations/collection/base.py (hunk 1 above)
Comment (TonyBagnall) [reply aeon/transformations/collection/base.py:396]: it is not called anywhere and is not tested

File: aeon/transformations/collection/_segment.py (hunk 2 above)
Comment (TonyBagnall) [reply aeon/transformations/collection/_segment.py:65]: this tag is not used anywhere and seems unnecessary to me, we can keep it if you want

Comment (TonyBagnall) [reply aeon/transformations/collection/_segment.py]: put back in

File: aeon/transformations/collection/base.py (hunk 1 above)
Comment (TonyBagnall) [reply aeon/transformations/collection/base.py:396]: put it back in, needs testing though

File: aeon/transformations/collection/_segment.py (hunk 2 above)
Comment (MatthewMiddlehurst) [reply aeon/transformations/collection/_segment.py]: Used in pipelines. Would leave removing for another PR

File: aeon/transformations/collection/base.py (hunk 1 above)
Comment (MatthewMiddlehurst) [reply aeon/transformations/collection/base.py:396]: Thats pretty weird, kind of implies that we arent testing `y` for these transformers.

**Review by MatthewMiddlehurst — APPROVED** [review]

**Pull request merged**